url11fb0973cf_raichuman9_Swimmer526_tgz-pJ8-fr_inria_optimization_cmaes_examples_CMAExample1J8/src/fr/inria/optimization/cmaes/CMAEvolutionStrategy.java

```diff
@@ -0,0 +1,2704 @@
+//                    parseDouble(getAllToken(properties.getProperty("boundaryUpper"))));
+        if ((s = properties.getProperty("typicalX")) != null) {
+            setTypicalX(options.parseDouble(options.getAllToken(s))); 
+        }
+        if ((s = properties.getProperty("initialX")) != null) {
+            setInitialX(options.parseDouble(options.getAllToken(s))); 
+        }
+        if ((s = properties.getProperty("initialStandardDeviations")) != null) {
+            setInitialStandardDeviations(options.parseDouble(options.getAllToken(s)));
+        }
+        if ((s = properties.getProperty("dimension")) != null) { // parseInt does not ignore trailing spaces
+            setDimension(Integer.parseInt(options.getFirstToken(s)));
+        }
+        if ((s = properties.getProperty("randomSeed")) != null) {
+            setSeed(Long.parseLong(options.getFirstToken(s)));
+        }
+        if ((s = properties.getProperty("populationSize")) != null) {
+            sp.setPopulationSize(Integer.parseInt(options.getFirstToken(s)));
+        }
+        if ((s = properties.getProperty("cCov")) != null) {
+            sp.setCcov(Double.parseDouble(options.getFirstToken(s)));
+        }
+        
+    }
+//    private void infoVerbose(String s) {
+//        println(" CMA-ES info: " + s);
+//    }
+    
+    private void warning(String s) {
+        println(" CMA-ES warning: " + s);
+    }
+    private void error(String s) { // somehow a relict from the C history of this code
+        println(" CMA-ES error: " + s);
+        //e.printStackTrace();            // output goes to System.err
+        //e.printStackTrace(System.out);  // send trace to stdout
+
+        throw new CMAException(" CMA-ES error: " + s);
+        //      System.exit(-1); 
+    }
+    
+    /** some simple math utilities */
+    class MyMath { // implements java.io.Serializable {
+        int itest;
+        
+        double square(double d) {
+            return d*d;
+        }
+        double prod(double []ar) {
+            double res = 1.0;
+            for(int i = 0; i < ar.length; ++i)
+                res *= ar[i];
+            return res;
+        }
+        
+        public double median(double ar[]) {
+            // need a copy of ar
+            double [] ar2 = new double[ar.length];
+            for (int i = 0; i < ar.length; ++i)
+                ar2[i] = ar[i];
+            Arrays.sort(ar2);
+            if (ar2.length % 2 == 0)
+                return (ar2[ar.length/2] + ar2[ar.length/2-1]) / 2.;
+            else    
+                return ar2[ar.length/2];
+        }
+        
+        /** @return Maximum value of 1-D double array */
+        public double max(double ar[]) {
+            int i;
+            double m;
+            m = ar[0];
+            for (i = 1; i < ar.length; ++i) {
+                if (m < ar[i])
+                    m = ar[i];
+            }
+            return m;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 907:912, 974:979.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.